service/matching/task_queue_partition_manager.go

```diff
@@ -115,18 +117,26 @@ func newTaskQueuePartitionManager(
 		)
 	}
 
-	unload := func(bool) {
+	unload := func() {
 		pm.unloadFromEngine(unloadCauseConfigChange)
 	}
 
 	var fairness bool
-	fairness, pm.cancelFairnessSub = tqConfig.EnableFairnessSub(unload)
+	changeKey := []byte(pm.partition.RoutingKey())
+	fairnessChangeChanged := makeChangeChangedFunc(changeKey, &fairness, unload)
+	fairnessChange, cancel1 := tqConfig.EnableFairnessSub(fairnessChangeChanged)
+	fairness = fairnessChange.Value(changeKey, time.Now())
+	pm.cancelFairnessSub = cancel1
```

Comment:
I think it makes sense to extract this into a helper that can enable you saying the same thing as before, but giving it the changeKey too:

```
fairness, pm.cancelFairnessSub = tqConfig.EnableFairnessSub(changeKey, unload)
```

Doesn't have to be in this PR though.